### DISCUSSION TITLE
format of word tables from quarto

### DISCUSSION THREAD

**Discussion by KipBalkcom-USDA** [quarto-dev/quarto-cli#9650] [Q&A]
### Description

````qmd
---
title: "Reproducible Table in Quarto"
engine: jupyter
format: 
  docx: 
    reference-doc: custom-reference-doc.docx
---

This is a minor format change, but different from the previous version of Quarto I was using (1.3.???). Notice the table output in word file (current_behavior.docx) appears to be two separate objects. Line numbers no longer include the table caption and a space exists between the table caption and table compared to the previous word file (previous_behavior.docx). 

```{python}
#| label: tbl-dates
#| tbl-cap: Years and locations.
#| echo: FALSE

import pandas as pd
from tabulate import tabulate
from IPython.display import Markdown

# define values
field_ops = (pd.DataFrame({
    "Season": ['2020', '2021'],
    "Location": ["Loc1", "Loc2"]}))

Markdown(tabulate(
    field_ops, headers="keys", showindex=False, tablefmt="pipe"))

``` 
Current software:
Windows 10
Quarto 1.4.554
Word Microsoft 365
````
[custom-reference-doc.docx](https://github.com/quarto-dev/quarto-cli/files/15295004/custom-reference-doc.docx)
[current_behavior.docx](https://github.com/quarto-dev/quarto-cli/files/15295101/current_behavior.docx)
[previouis_behavior.docx](https://github.com/quarto-dev/quarto-cli/files/15295493/previouis_behavior.docx)

**Comment by mcanouil** [comment]
Could you put your question outside of your code?
Additionally, could you test without a `reference-doc` set as the issue might come from it.
Finally, could you try to remove `#| label: tbl-dates`? Do you see the same behaviour?

Note that Quarto updates Pandoc in each stable releases which might cause some changes in formats.

**Reply by KipBalkcom-USDA** [reply]
As suggested, I removed the reference-doc, which seemed to eliminate the extra space between the caption and the table. In the next step, I also removed the `#| label: tbl-dates` from the code.  This enabled the table caption to become part of the word document and line numbers are recognized for the table caption.

However, those changes eliminate the cross reference for the table and the format of the table caption text is not correct. 
Is there a way to use the cross referencing and format the caption so I don't have to do that manually?

[current_behavior.docx](https://github.com/quarto-dev/quarto-cli/files/15296049/current_behavior.docx)

Current software:
Windows 10
Quarto 1.4.554
Word Microsoft 365

````qmd
---
title: "Reproducible Table in Quarto"
engine: jupyter
format: docx 
---

```{python}
#| tbl-cap: Years and locations.
#| echo: FALSE

import pandas as pd
from tabulate import tabulate
from IPython.display import Markdown

# define dates
field_ops = (pd.DataFrame({
    "Year": ['2020', '2021'],
    "Location": ["Loc1", "Loc2"]}))

Markdown(tabulate(
    field_ops, headers="keys", showindex=False, tablefmt="pipe"))

``` 
````

**Reply by cscheid** [reply]
Does the problem persist if you add `label: tbl-mytable` to the code cell metadata? In general, captions need corresponding ids in order for the crossref system to find them.

**Reply by mcanouil** [reply]
The user went ahead and opened an issue:

- https://github.com/quarto-dev/quarto-cli/issues/9655

@KipBalkcom-USDA I wanted to kindly notify you to continue on an ongoing discussion within the existing thread before initiating a new one (even in the absence of reply). This approach ensures that our communication remains organised and efficient. Your cooperation is greatly appreciated.

**Reply by mcanouil** [reply]
Removing the label was to figure out what caused the "formatting issue".
Apparently, removing the label solved it, so it's the new cross-reference system which procuces a slightly different output.

**Reply by KipBalkcom-USDA** [reply]
My apologies, but there was a large icon on the right hand side of the page indicating for me to start an issue after I posted my response to you.  As someone who does not use this system very frequently, I thought that was your response, so I created the issue. Thanks for your understanding.
